waku/v2/protocol/waku_rln_relay/rln.nim

```diff
@@ -29,38 +29,61 @@ type Auth* = object
   index*: uint
   
 #------------------------------ Merkle Tree operations -----------------------------------------
-
 proc update_next_member*(ctx: RLN[Bn256],
                         input_buffer: ptr Buffer): bool {.importc: "update_next_member".}
+## input_buffer points to the id commitment byte seq
+## the return bool value indicates the success or failure of the operation
 
 proc delete_member*(ctx: RLN[Bn256], index: uint): bool {.importc: "delete_member".}
+## index is the position of the id commitment key to be deleted from the tree
+## the return bool value indicates the success or failure of the operation
 
 proc get_root*(ctx: RLN[Bn256], output_buffer: ptr Buffer): bool {.importc: "get_root".}
-
+## get_root populates the passed pointer output_buffer with the current tree root
+## the output_buffer holds the Merkle tree root of size 32 bytes
+## the return bool value indicates the success or failure of the operation
+## 
 #----------------------------------------------------------------------------------------------
 #-------------------------------- zkSNARKs operations -----------------------------------------
-
 proc key_gen*(ctx: RLN[Bn256], keypair_buffer: ptr Buffer): bool {.importc: "key_gen".}
+## generates id key and id commitment key serialized inside keypair_buffer as | id_key <32 bytes>| id_commitment_key <32 bytes> |
+## the return bool value indicates the success or failure of the operation
 
 proc generate_proof*(ctx: RLN[Bn256],
                     input_buffer: ptr Buffer,
                     output_buffer: ptr Buffer): bool {.importc: "generate_proof".}
 ## input_buffer serialized as  [ id_key<32> | id_index<8> | epoch<32> | signal_len<8> | signal<var> ]
```

Comment:
Why `id_index` is 8 bytes long? Aren't 4 enough? (32 bytes * 2^(8*4) = 128GiB of leaves).
I guess its based on the typical 8-bytes `usize` for most 64bits architecture, but over 32bits architectures such length won't make sense.

Reply:
`uint8` is the smallest among the integer types supported by Rust (and typically by other languages), which also appeared to be large enough for the leaf index. Of course, smaller than that also works but requires a custom type definition.

Reply:
Right, misinterpreted the sizes. Right, the 32 depth of the tree seems reasonable and large enough, and perhaps `uint32` would also do the job. But considering that this data is not being transmitted as part of the proof on the wire, it does not cost bandwidth. Would it be of any help to change it to `uint32`? if yes, then we can do it.

Reply:
The problem here is not about bandwidth but the architecture: on 32 bits machines `usize` is 4 bytes and you then need a safe conversion from `u64` to `usize` in order to pass this to the RLN API. The other way around, i.e. when serializing the input, keeping index at 8 bytes, requires casting to a non-native type (`u64`) before serializing it. This extra overhead and extra checks doesn't make sense here if 4 bytes out of 8 will always be 0. We can address this in a later PR.

Reply:
I see, if the concern is about interaction with the Rust lib API, then the rln lib and the nim code will run and be compiled on the same machine with the same architecture, right? then my understanding is that `uint` will be rendered with an identical size both by the nim code and by the underlying lib, so no conversion would be needed. Or are you referring to some other issue?

Reply:
The issue will be then in the deserialization: right now you read a fixed amount of 8 bytes (the common `usize` byte-size over 64 bits architectures); to make doable what you say you need to first get the size of `usize` (4 or 8, but not necessarily) and then read that amount of bytes to deserialize the index. To me makes sense to just work with `u32`.